### PR TITLE
Making SideNavTopBar importable and editable

### PR DESCRIPTION
**Pull request by yale**

### CODE REVIEW

File: dist/styles.css [hunk 1]
```diff
@@ -1701,11 +1701,16 @@
...
 
-.src-components-SideNavBar-top__1PMat {
+.src-components-SideNavBar-top__1PMat,
+.src-components-SideNavBar-input__3y-7h {
   font-size: 14px;
+  color: white;
+  font-weight: bold;
+}
+
+.src-components-SideNavBar-top__1PMat {
```
Comment (GeReV) [inline dist/styles.css]: How did we end up with these horrible class names? :\

File: src/components/SideNavBar/SideNavBar.jsx [hunk 2]
```diff
@@ -20,46 +20,54 @@ type Props = {|
...
-        key={item.id}
-        id={item.id}
-        icon={item.icon}
-        label={item.label}
-        onClick={this.handleClick}
-        items={item.items}
-        selectedId={this.props.selectedId}
-      />)
+    return (
+      this.props.items &&
```
Comment (GeReV) [inline src/components/SideNavBar/SideNavBar.jsx]: `items` is defined as `Array<Item>`, so no need to test them.

File: src/components/SideNavBar/SideNavBar.jsx [hunk 3]
```diff
@@ -20,46 +20,54 @@ type Props = {|
...
-        <SideNavBarTop
-          label={this.props.topText}
-          icon={this.props.topIcon}
-          onClick={this.props.onClickTop}
-        />
-        <div className={style.sidebarContent}>
-          {this.renderItems()}
-        </div>
+      <div {...getDataAttrs(this.props.data)} className={style.wrapper}>
+        {(this.props.topText || this.props.topIcon) && (
```
Comment (GeReV) [inline src/components/SideNavBar/SideNavBar.jsx]: Destructure `this.props`?

File: src/components/SideNavBar/SideNavBarTop.jsx [hunk 4]
```diff
@@ -8,27 +8,71 @@ type Props = {|
...
   icon: string,
   onClick: () => void,
+  editing: boolean,
+  placeholder: string,
+  value: string,
+  onChange: ({ value: string }) => void,
+  onBlur: () => void,
+  fullWidth: boolean,
+  className: string,
+  forwardedRef?: { current: null | HTMLInputElement },
```
Comment (GeReV) [inline src/components/SideNavBar/SideNavBarTop.jsx]: I believe there is a `React.Ref` type.

File: src/components/SideNavBar/SideNavBarTop.jsx [hunk 5]
```diff
@@ -8,27 +8,71 @@ type Props = {|
...
-            icon={icon}
-          />
-        </div>
-        <div className={cn(style.groupLabelWrapper, style.topText)}>
-          {label}
+class SideBarNavTop extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props);
+    // $FlowFixMe (https://github.com/facebook/flow/issues/6103)
+    this.input = props.forwardedRef || React.createRef();
```
Comment (GeReV) [inline src/components/SideNavBar/SideNavBarTop.jsx]: You can move this out of the constructor:
```jsx
this.input = this.props.forwardedRef || React.createRef();
```

File: src/components/SideNavBar/style.scss [hunk 6]
```diff
@@ -139,3 +163,27 @@
...
+  &:focus {
+    outline: none;
+    color: $cGray20;
+  }
+
+  &::-webkit-input-placeholder,
+  &::placeholder,
+  &[placeholder] {
+    text-overflow: ellipsis;
+    white-space: nowrap;
```
Comment (GeReV) [inline src/components/SideNavBar/style.scss]: Might need `overflow: hidden`.

File: src/components/SideNavBar/SideNavBarTop.jsx (hunk 4 above)
Comment (yale) [reply src/components/SideNavBar/SideNavBarTop.jsx]: Until the upcoming version of flow, that type doesn't reflect react 16's new createRef API (e.g., the ref does not have a `current` attribute). This is the recommended workaround.

File: dist/styles.css [hunk 7]
```diff
@@ -1701,11 +1701,16 @@
...
   box-sizing: border-box;
   background-color: #2b3445;
-  position: absolute;
   left: 0;
```
Comment (pseuyi) [inline dist/styles.css]: do we still need `left`, `right`, and `top`?

File: stories/SideNavBar.js [hunk 8]
```diff
@@ -14,12 +14,38 @@ class StatefulWrapper extends React.Component {
...
+class StatefulWrapperWithEditableTop extends React.Component {
+  constructor() {
+    super();
+    this.state = { selectedId: null, editingTopText: false };
+  }
+
+  render() {
+    return (
+      <StatefulWrapper
+        onClickTop={event => {
```
Comment (pseuyi) [inline stories/SideNavBar.js]: prefer to extract these inline fns

**Pull request merged**